mdata/store_cassandra.go

```diff
@@ -77,9 +78,37 @@ type cassandraStore struct {
 	session     *gocql.Session
 	writeQueues []chan *ChunkWriteRequest
 	readQueue   chan *ChunkReadRequest
+	ttlTables   map[uint32]string
 }
 
-func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password, hostSelectionPolicy string, timeout, readers, writers, readqsize, writeqsize, retries, protoVer int, ssl, auth, hostVerification bool) (*cassandraStore, error) {
+func (c *cassandraStore) ensureTtlTables(ttls []uint32, keyspace string) error {
+	for _, ttl := range ttls {
+		// calculate table name and window
+		var window int = 1
+		for tmp := ttl / (24 * 60 * 60); tmp > 1; tmp = tmp / 2 {
+			window *= 2
+		}
+		table_name := fmt.Sprintf("metric_%d", window)
+
+		// table isn't know to exist, ensure it exists
+		err := c.session.Query(fmt.Sprintf(table_schema, keyspace, table_name, window)).Exec()
```

Comment:
That's a really interesting discussion. I think for me the reason why I decided to add the option was simple: 

I'm pretty sure we will never need to change that to another value than the default `20` in our use case, but since this available online other people who have different use cases might use it. Can we be sure that there is not a single possible use case in which it makes sense to tune that value? I don't think so. So I added it because it's not much additional overhead anyway.